src/lib/Menu/ContentRightSidebarBuilder.php

```diff
@@ -189,11 +178,29 @@ public function createStructure(array $options): ItemInterface
         ];
         $copySubtreeAttributes = [
             'class' => 'ez-btn--udw-copy-subtree',
-            'data-udw-config' => $this->udwExtension->renderUniversalDiscoveryWidgetConfig('single_container'),
-            'data-root-location' => $this->configResolver->getParameter(
-                'universal_discovery_widget_module.default_location_id'
-            ),
+            'data-udw-config' => $uwdConfig,
+            'data-root-location' => $rootLocation,
+        ];
+        $moveAttributes = [
+            'class' => 'btn--udw-move',
+            'data-udw-config' => $uwdConfig,
+            'data-root-location' => $rootLocation,
+        ];
+        $copyAttributes = [
+            'class' => 'btn--udw-copy',
+            'data-udw-config' => $uwdConfig,
+            'data-root-location' => $rootLocation,
         ];
+        $createPolicies = $this->permissionResolver->hasAccess(
+            'content',
+            'create'
+        );
+        $manageLocationsPolicies = $this->permissionResolver->hasAccess(
+            'content',
+            'manage_locations'
+        );
+        $hasCreatePermission = !is_bool($createPolicies) || $createPolicies;
+        $hasManageLocationsPermission = !is_bool($manageLocationsPolicies) || $manageLocationsPolicies;
```

Comment:
This logic could be extracted to the dedicated function to reduce complexity and readability